usr/share/lib/img_proof/tests/SLES/EC2/test_sles_ec2_dracut_conf.py

```diff
@@ -5,34 +5,43 @@ def test_sles_ec2_dracut_conf(host, get_release_value, determine_architecture):
     if determine_architecture() != 'X86_64':
         pytest.skip('Only x86_64 architecture is tested.')
 
-    needed_drivers = (
-        'ena',
-        'nvme',
-        'nvme-core',
-        'virtio',
-        'virtio_scsi',
-        'xen-blkfront',
-        'xen-netfront'
-    )
+    needed_drivers = {
+        '07-aws-type-switch.conf': [
+            'ena',
+            'virtio',
+            'virtio_scsi',
+            'xen-blkfront',
+            'xen-netfront'
+        ],
+        '07-nvme.conf': [
+            'nvme',
+            'nvme-core'
+        ]
+    }
+
     version = get_release_value('VERSION')
     assert version
 
     name = get_release_value('PRETTY_NAME').lower()
     assert name
 
-    dracut_conf = host.file('/etc/dracut.conf.d/07-aws-type-switch.conf')
-
-    assert dracut_conf.exists
-    assert dracut_conf.is_file
-
-    for driver in needed_drivers:
-        if (
-            driver.startswith('virtio') and
-            (
-                version.startswith('15') or
-                version == '12-SP5' or
-                'micro' in name
-            )
-        ):
-            continue
-        assert dracut_conf.contains(driver)
+    base_dracut_config_dir = '/etc/dracut.conf.d/'
```

Comment:
Looks good. We'll need to change this eventually when the next package versions are released.

Reply:
yep, we'll need to adjust the base config dir and the expected dictionary when other changes make it to the pipeline.
We can leave this PR open and merge it once we've got the final version.

Reply:
A couple images apparently where missing the `07-nvme.conf` expected file, so I modified the test to scan all the files in the config dir to look for the expected modules.

Reply:
And we want to expand this. The nvme configuration is expected for all frameworks.

Reply:
> A couple images apparently where missing the `07-nvme.conf` expected file, so I modified the test to scan all the files in the config dir to look for the expected modules.

Probably some images are not updated yet with latest builds.

Reply:
> And we want to expand this. The nvme configuration is expected for all frameworks.

If this is the case then we probably want to split up the test. One for the aws type switch drivers and one for the nvme drivers.